### PR TITLE
chore(node): prefer node:* imports internally

### PR DESCRIPTION
**Pull request by SimonSchick**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

This is basically https://github.com/DefinitelyTyped/DefinitelyTyped/pull/51107 (minus leaving older versions in place).

This is strictly an internal change and should *not* affect consumers, I noticed that when running `tsc --listFiles` the `assert` package was included rather than the node module, forcing node typings to use the prefix internally fixes this by disambiguating the modules.

### CODE REVIEW

**Comment by SimonSchick** [comment]
Also renamed `diagnostic_channel` to `diagnostics_channel`, made that mistake here.

File: package.json [hunk 1]
```diff
@@ -28,8 +28,8 @@
         "@definitelytyped/definitions-parser": "latest",
         "@definitelytyped/dtslint-runner": "latest",
         "@definitelytyped/utils": "latest",
-        "danger": "^10.1.1",
         "@octokit/rest": "^16.0.0",
+        "danger": "^10.1.1",
```
Comment (elibarzilay) [inline package.json:32]: redundant change?

Comment (SimonSchick) [reply package.json:32]: Eh, npm sorted the depedencies when I re-installed the dev deps, not rly worth reverting this change, builds take a long time.

File: types/aws-serverless-express/middleware.d.ts [hunk 2]
```diff
@@ -1,24 +1,22 @@
...
-type NonFunctionProperties<T> = Pick<T, NonFunctionPropertyNames<T>>;
-
-declare module 'http' {
-  interface IncomingMessage {
-    apiGateway?: {
-      event: Omit<APIGatewayProxyEvent, 'body'>;
-      context: NonFunctionProperties<Context>;
-    } | undefined;
-  }
+declare module 'node:http' {
```
Comment (peterblazejewicz) [inline types/aws-serverless-express/middleware.d.ts:5]: @SimonSchick same question as in other pr, does it introduce Node v16 dependency?

File: package.json (hunk 1 above)
Comment (elibarzilay) [reply package.json:32]: (Yeah, I saw that later in local tests)

**Comment by SimonSchick** [comment]
@peterblazejewicz I can't seem to get it to work augmenting both `http` and `node:http` modules, going to need some help here.

**Comment by peterblazejewicz** [comment]
> @peterblazejewicz I can't seem to get it to work augmenting both `http` and `node:http` modules, going to need some help here.

I've already asked the question on discord, tbu

**Comment by SimonSchick** [comment]
@peterblazejewicz I had to revert some of the changes as they caused compatibility issues but the core intent is still there, `tsc --listFiles` no longer includes non `node` and TSLib files.

**Comment by SimonSchick** [comment]
Anyways, all resolved now, I think we can move forward w/ this and we can move towards adding the auto generated documentation.

**Review by peterblazejewicz — APPROVED** [review]
LGMT!

**Comment by SimonSchick** [comment]
Ready to merge

**Pull request merged**